pkg/operation/botanist/waiter_test.go

```diff
@@ -48,10 +50,40 @@ var _ = Describe("Waiter", func() {
 			Operation: &operation.Operation{
 				Logger:         logr.Discard(),
 				ShootClientSet: shootClientSet,
+				Shoot:          &shootpkg.Shoot{},
 			},
 		}
 	})
 
+	Describe("#WaitUntilNodesDeleted", func() {
+		var (
+			node *corev1.Node
+		)
+
+		BeforeEach(func() {
+			node = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "infinity-node",
+				},
+			}
+		})
+
+		It("should return ok when no node is found", func() {
+			Expect(botanist.WaitUntilNodesDeleted(ctx)).To(Succeed())
+		})
+
+		It("should return an error when a node is still present", func() {
+			ctxCanceled, cancel := context.WithCancel(ctx)
+			cancel()
+
+			Expect(botanist.ShootClientSet.Client().Create(ctxCanceled, node)).To(Succeed())
```

Comment:
```suggestion
			Expect(botanist.ShootClientSet.Client().Create(ctx, node)).To(Succeed())
```

To emphasize that the cancelled context is only required for the tested function. Same for other occurrences in this file.

Reply:
Fixed